client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -348,8 +389,42 @@ func (d *DockerOps) isFileServiceRunning(servID string, volName string) (uint32,
 //      bool:    The result of the operation. True if the service was
 //               successfully stopped.
 func (d *DockerOps) StopSMBServer(volName string) (int, string, bool) {
-	log.Errorf("stopSMBServer to be implemented")
-	return 0, "", true
+	serviceID, _, err := d.getServiceIDAndPort(volName)
+	if err != nil {
+		return 0, "", false
+	}
+
+	//Stop the service
+	err = d.Dockerd.ServiceRemove(context.Background(), serviceID)
+	if err != nil {
+		log.Warningf("Failed to remove file server for volume %s. Reason: %v",
+			volName, err)
+		return 0, "", false
+	}
+
+	// Wait till service container stops
+	ticker := time.NewTicker(checkDuration)
+	defer ticker.Stop()
+	timer := time.NewTimer(sambaRequestTimeout)
+	defer timer.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			log.Infof("Checking status of file server container...")
+			serviceID, _, err := d.getServiceIDAndPort(volName)
+			if err != nil && !strings.Contains(err.Error(), "No service returned") {
+				return 0, "", false
+			}
+			// service is removed successfully
+			if serviceID == "" {
+				return 0, "", true
+			}
+		case <-timer.C:
+			log.Warningf("Timeout reached while waiting for file server container for volume %s",
```

Comment:
Nit: "Timeout reached while waiting for file server container for volume %s **to stop**." Lets make sure log message is different from the start function message.